File: src/main/java/employeetech/com/employee/manager/service/DepartmentService.java
package employeetech.com.employee.manager.service;

import employeetech.com.employee.manager.repo.DepartmentRepo;

public class DepartmentService {

    public DepartmentRepo departmentRepo;

    public DepartmentService( DepartmentRepo departmentRepo) {
        this.departmentRepo = departmentRepo;
    }
}

File: src/main/kotlin/employeetech/com/employee/manager/service/DepartmentService.kt
package employeetech.com.employee.manager.service

import employeetech.com.employee.manager.model.Department
import employeetech.com.employee.manager.repo.DepartmentRepo
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class DepartmentService(private val departmentRepo: DepartmentRepo) {

    fun addDepartment(department: Department): Department {
        return departmentRepo.save(department)
    }

    fun findAllDepartments(): List<Department> {
        return departmentRepo.findAll()
    }

    fun updateDepartment(department: Department): Department {
        return departmentRepo.save(department)
    }

    fun deleteDepartment(id: Long) {
        departmentRepo.deleteById(id)
    }
}
